router/sources/router.move

```diff
@@ -0,0 +1,928 @@
+        let router_config = borrow_global_mut<RouterConfig>(@router);
+        assert!(router_config.admin_addr == address_of(router_admin), error::permission_denied(ENOT_ADMIN));
+        router_config.mode = mode;
+    }
+
+    // == ROUTER MANAGEMENT READ FUNCTIONS ==
+
+    inline fun get_router_signer(): signer acquires RouterConfig {
+        let router_config = borrow_global<RouterConfig>(@router);
+        account::create_signer_with_capability(&router_config.signer_cap)
+    }
+
+    inline fun router_signer_addr(): address acquires RouterConfig {
+        let router_config = borrow_global<RouterConfig>(@router);
+        account::get_signer_capability_address(&router_config.signer_cap)
+    }
+
+    inline fun is_valid_mode(mode: u8): bool {
+        mode <= MODE_V1_AND_V2
+    }
+
+    #[view]
+    public fun get_admin_addr(): address acquires RouterConfig {
+        let router_config = borrow_global<RouterConfig>(@router);
+        router_config.admin_addr
+    }
+
+    #[view]
+    public fun get_pending_admin_addr(): Option<address> acquires RouterConfig {
+        let router_config = borrow_global<RouterConfig>(@router);
+        router_config.pending_admin_addr
+    }
+
+    #[view]
+    public fun get_mode(): u8 acquires RouterConfig {
+        let router_config = borrow_global<RouterConfig>(@router);
+        router_config.mode
+    }
+
+    // == ROUTER WRITE FUNCTIONS ==
+
+    // ==== REGISTRATION ====
+
+    /// If the name is registered and active in v1, then the name can only be registered if we have burned the token (sent it to the router_signer)
+    /// Else, the name can only be registered if it is available in v2 (we double check availablity for safety)
+    inline fun can_register_in_v2(domain_name: String, subdomain_name: Option<String>): bool {
+        if (domains::name_is_registered(
+            subdomain_name,
+            domain_name
+        ) && !domains::name_is_expired(subdomain_name, domain_name)) {
+            let (is_burned, _token_id) = domains::is_token_owner(
+                router_signer_addr(),
+                subdomain_name,
+                domain_name
+            );
+            is_burned
+        } else {
+            v2_domains::is_name_registerable(domain_name, subdomain_name)
+        }
+    }
+
+    /// @param user The user who is paying for the registration
+    /// @param domain_name The domain name to register
+    /// @param registration_duration_secs The duration of the registration in seconds
+    /// @param target_addr The address the registered name will point to
+    /// @param to_addr The address to send the token to. If none, then the user will be the owner. In MODE_V1, receiver must have already opted in to direct_transfer via token::opt_in_direct_transfer
+    public entry fun register_domain(
+        user: &signer,
+        domain_name: String,
+        registration_duration_secs: u64,
+        target_addr: Option<address>,
```

Comment:
Is it possible to simplify here and always require users to specify target and to addresses? They can also way set to user's address initially and change it later if needed. This would greatly simplify the code + reduce branches. Also makes the function easier to call not having to deal with options